code/modules/food_and_drink/snacks.dm

```diff
@@ -1216,25 +1224,44 @@
 			user.show_text("The [tube] isn't full enough to add frosting.", "red")
 			return
 
-		if (src.style_step > length(src.styles))
+		if (src.style_step > 2) // only allow up to two frosting types on a single donut
 			user.show_text("You can't add anymore frosting.", "red")
 			return
 
+		var/frostingtype = null
+		frostingtype = input("Which frosting style would you like?", "Frosting Style", null) as null|anything in frostingstyle
+		if(frostingtype && (get_dist(src, user) <= 1))
+			switch(frostingtype)
+				if("icing")
+					frostingtype = "icing"
+				if("half and half icing")
+					frostingtype = "half"
+				if("dipped")
+					frostingtype = "dipped"
+				if("center fill")
+					frostingtype = "center"
+				if("zigzags")
+					frostingtype = "zigzags"
+				if("star")
+					frostingtype = "star"
+				if("heart")
+					frostingtype = "heart"
+			if(!src.GetOverlayImage(frostingstyle, src.layer + 0.2))
+				var/frostingstyle[src.style_step]
```

Comment:
I'm unsure what you are even trying to do on this line, I think cou can remove it considering my next suggestion?

Reply:
Ty for the reviews; I will update the frosting variable name and remove this in a little bit once I can switch branches again!